app/src/main/java/ch/epfl/sdp/blindly/user/User.kt

```diff
@@ -0,0 +1,134 @@
+package ch.epfl.sdp.blindly.user
+
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.ktx.getField
+import kotlinx.serialization.Serializable
+import java.time.LocalDate
+import java.time.Period
+
+/**
+ * A class used to represent a user
+ */
+@Serializable
+class User private constructor(
+                val username: String?,
+                val location: String?,
+                val birthday: String?,
+                val gender: String?,
+                val sexual_orientations: List<String>,
+                val show_me: String?,
+                val passions: List<String>,
+                val radius: Int?,
+                val matches: List<User>,
+                val description: String?) {
+
+    /**
+     * A builder used to partially initialize a user during the profile_setup activities
+     * Made serializable so that it can be put in a bundle and passed as extrax
+     */
+    @Serializable
+    data class Builder(
+        var username: String? = null,
+        var location: String? = null,
+        var birthday: String? = null,
+        var gender: String? = null,
+        var sexual_orientations: List<String> = listOf(),
+        var show_me: String? = null,
+        var passions: List<String> = listOf(),
+        var radius: Int? = null,
+        var matches: List<User> = listOf(),
+        var description: String? = null) {
+
+        fun setUsername(username: String) = apply {
+            this.username = username
+        }
+        fun setLocation(location: String) = apply {
+            this.location = location
+        }
+        fun setBirthday(birthday: String) = apply {
+            this.birthday = birthday
+        }
+        fun setGender(gender: String) = apply {
+            this.gender = gender
+        }
+        fun setSexualOrientations(sexual_orientations: List<String>) = apply {
+            this.sexual_orientations = sexual_orientations
+        }
+        fun setShowMe(showMe: String) = apply {
+            this.show_me = showMe
+        }
+        fun setPassions(passions: List<String>) = apply {
+            this.passions = passions
+        }
+        fun setRadius(radius: Int) = apply {
+            this.radius = radius
+        }
+        fun setMatches(matches: List<User>) = apply {
+            this.matches = matches
+        }
+        fun setDescription(description: String) = apply {
+            this.description = description
+        }
```

Comment:
Aren't all those functions already provided by default in kotlin ? (`setUsername(newUserName)` would be `.username = newUserName`).

Reply:
No, I think it's a bit different with Builders inside a class (and I've just tried it to be safe but .username = newUserName is flagged as an undefined ref)

Reply:
Okay my bad then.